docs/pages/build-reference/e2e-tests.mdx

```diff
@@ -1,763 +1,166 @@
 
-function cleanup()
-{
-  echo 'Cleaning up...'
-  if [[ "$EAS_BUILD_PLATFORM" == "android" ]]; then
-    # Kill emulator
-    adb emu kill &
-  fi
-}
+To execute a custom build using the `build-and-maestro-test` profile that will build your app and run the Maestro E2E tests afterward, run the following command:
 
-if [[ "$EAS_BUILD_PROFILE" != "test"* ]]; then
-  exit
-fi
-
-# Fail if anything errors
-set -eox pipefail
-# If this script exits, trap it first and clean up the emulator
-trap cleanup EXIT
-
-ANDROID_EMULATOR=pixel_4
-
-if [[ "$EAS_BUILD_PLATFORM" == "android" ]]; then
-  # Start emulator
-  $ANDROID_SDK_ROOT/emulator/emulator @$ANDROID_EMULATOR -no-audio -no-boot-anim -no-window -use-system-libs 2>&1 >/dev/null &
-
-  # Wait for emulator
-  max_retry=10
-  counter=0
-  until adb shell getprop sys.boot_completed; do
-    sleep 10
-    [[ counter -eq $max_retry ]] && echo "Failed to start the emulator!" && exit 1
-    counter=$((counter + 1))
-  done
-
-
-  # Execute Android tests
-  if [[ "$EAS_BUILD_PROFILE" == "test" ]]; then
-    detox test --configuration android.release
-  fi
-  if [[ "$EAS_BUILD_PROFILE" == "test_debug" ]]; then
-    detox test --configuration android.debug
-  fi
-else
-  # Execute iOS tests
-  if [[  "$EAS_BUILD_PROFILE" == "test" ]]; then
-    detox test --configuration ios.release
-  fi
-  if [[ "$EAS_BUILD_PROFILE" == "test_debug" ]]; then
-    detox test --configuration ios.debug
-  fi
-fi
-```
+<Terminal cmd={['$ npx eas-cli@latest build --profile build-and-maestro-test']} />
 
-</Collapsible>
+<ContentSpotlight
+  alt="Logs of successful Maestro tests"
+  src="/static/images/eas-build/tests/03-logs.png"
+  className="max-w-[480px]"
+/>
 
-<Collapsible summary="eas.json">
+Any Maestro artifacts including screenshots taken when flow fails that are saved at `/Users/expo/.maestro/tests` (the default destination) will be automatically uploaded as build artifacts. You can download them from the build page.
 
-```json eas.json
-{
-  "build": {
-    "test": {
-      "android": {
-        "gradleCommand": ":app:assembleRelease :app:assembleAndroidTest -DtestBuildType=release",
-        "withoutCredentials": true
-      },
-      "ios": {
-        "simulator": true
-      }
-    },
-    /* @info New section */ "test_debug": {
-      "android": {
-        "gradleCommand": ":app:assembleDebug :app:assembleAndroidTest -DtestBuildType=debug",
-        "withoutCredentials": true
-      },
-      "ios": {
-        "buildConfiguration": "Debug",
-        "simulator": true
-      },
-      "env": {
-        "EXPO_USE_UPDATES": "1"
-      },
-      "channel": "test_debug"
-    } /* @end */
-  }
-}
-```
+<ContentSpotlight
+  alt="Failed Maestro tests artifacts"
+  src="/static/images/eas-build/tests/04-artifacts.png"
+  className="max-w-[480px]"
+/>
+</Step>
 
-</Collapsible>
+If you want to build more advanced custom builds workflows check the [custom build schema reference](/custom-builds/schema) for more information.
```

Comment:
```suggestion
## More

If you want to build more advanced custom builds workflows, see the [custom build schema reference](/custom-builds/schema/) for more information.

To learn more about Maestro flows and how to write them, see the [Maestro documentation](https://maestro.mobile.dev).
```

Let's add a section here to link to custom build schema doc and Maestro documentation.